ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -219,6 +253,85 @@ class ReplayKitVideoSource: NSObject, TVIVideoCapturer {
         }
         to.consumeCapturedFrame(frame)
         lastTimestamp = timestamp
+
+        // Frame retransmission logic.
+        if (ReplayKitVideoSource.retransmitLastFrame) {
+            if ReplayKitVideoSource.useDispatchQueue {
+                lastFrameStorage = frame
+                lastTransmitTimestamp = CMClockGetTime(CMClockGetHostTimeClock())
+                dispatchRetransmissions()
+            } else {
+                lastFrame = frame
+                if retransmitTimer == nil {
+                    scheduleRetransmissions()
+                }
+            }
+        }
+    }
+
+    func scheduleRetransmissions() {
+        retransmitTimer?.invalidate()
+        let timer = Timer(timeInterval: ReplayKitVideoSource.kFrameRetransmitInterval,
+                                repeats: true,
+                                block: { (timer) in
+                                    if let frame = self.lastFrame,
+                                        let consumer = self.captureConsumer {
+                                        let currentTimestamp = CMClockGetTime(CMClockGetHostTimeClock())
+                                        let delta = CMTimeSubtract(currentTimestamp, frame.timestamp).seconds
+                                        let threshold = Double(4.0 / 60.0)
+
+                                        if delta >= threshold {
+                                            self.deliverFrame(to: consumer,
+                                                              timestamp: currentTimestamp,
+                                                              buffer: frame.imageBuffer,
+                                                              orientation:
+                                                frame.orientation)
+                                        }
+                                    }})
+        retransmitTimer = timer
+        RunLoop.main.add(timer, forMode: RunLoopMode.commonModes)
+    }
+
+    func dispatchRetransmissions() {
+        // This is a workaround for the fact that we can't provide our own serial queue to ReplayKit.
+        let currentQueue = ExampleCoreAudioDeviceGetCurrentQueue()
+
+        var source: DispatchSourceTimer?
+        if timerSource == nil {
+            source = DispatchSource.makeTimerSource(flags: DispatchSource.TimerFlags.strict,
+                                                    queue: currentQueue)
+            // Event handler.
+            source?.setEventHandler(handler: {
```

Comment:
We might suffer from clock drift when computing the timestamps for retransmitted frames. I used relative, rather than absolute deltas to compensate for the unknown delay of the rest of the ReplayKit XPC pipeline.